generator.go

```diff
@@ -0,0 +1,723 @@
+				}
+			}
+		}
+	}
+
+	if gregorianCalendar.DayPeriods != nil && gregorianCalendar.DayPeriods.DayPeriodContext != nil {
+		for _, periodContext := range gregorianCalendar.DayPeriods.DayPeriodContext {
+			if periodContext.Type != "format" {
+				continue
+			}
+
+			periods := map[string]string{}
+			for _, periodWidth := range periodContext.DayPeriodWidth {
+				if periodWidth.Type != "abbreviated" && periodWidth.Type != "narrow" {
+					continue
+				}
+
+				for _, period := range periodWidth.DayPeriod {
+					if p, ok := dayPeriodsStdMap[period.Type]; ok {
+						// preference for non-variant periods
+						if _, ok = periods[p]; ok && period.Alt == "variant" {
+							continue
+						}
+
+						periods[p] = strings.ReplaceAll(period.CharData, "\u202F", "")
+					}
+				}
+
+				if len(periods) == 2 {
+					locale.dayPeriods = periods
+					break
+				}
+			}
+		}
+	}
+
+	return nil
+}
+
+func lookupMonthValue(curr map[string]string, stdTab []string, lookupTable []*MonthWidth) (map[string]string, error) {
+	if curr == nil && len(lookupTable) == 0 {
+		return nil, nil
+	}
+
+	val := make(map[string]string, 12)
+	if curr != nil {
+		maps.Copy(val, curr)
+	}
+
+	for _, month := range lookupTable {
+		m, err := strconv.Atoi(month.Type)
+		if err != nil {
+			return nil, err
+		}
+
+		val[stdTab[m-1]] = month.CharData
+	}
+
+	return val, nil
+}
+
+func lookupDayValue(curr map[string]string, stdTab map[string]string, lookupTable []*Common) map[string]string {
+	if curr == nil && len(lookupTable) == 0 {
+		return nil
+	}
+
+	val := make(map[string]string, 7)
+	if curr != nil {
+		maps.Copy(val, curr)
+	}
+
+	for _, day := range lookupTable {
+		d, ok := stdTab[day.Type]
+		if !ok {
+			continue
+		}
+
+		val[d] = day.CharData
+	}
+
+	return val
+}
+
+func findGregorianCalendar(lang *LDML) *Calendar {
+	if lang == nil || lang.Dates == nil || lang.Dates.Calendars == nil || lang.Dates.Calendars.Calendar == nil {
+		return nil
+	}
+
+	for _, calendar := range lang.Dates.Calendars.Calendar {
+		if calendar.Type == "gregorian" {
+			return calendar
+		}
+	}
+
+	return nil
+}
+
+// WriteGoFile prepends a standard file comment and package statement to the
+// given bytes, applies gofmt, and writes them to a file with the given name.
+func writeGoFile(filename, pkg string, b []byte) error {
+	w, err := os.Create(filename)
+	if err != nil {
+		return err
+	}
+
+	defer w.Close()
+
+	header := "// Code generated by running \"go generate\" in github.com/elastic/lunes. DO NOT EDIT.\n\n"
+	if _, err = writeGo(w, pkg, header, b); err != nil {
+		return fmt.Errorf("error writing file %s: %w", filename, err)
+	}
+
+	return nil
+}
+
+// WriteGo prepends a standard file comment and package statement to the given
+// bytes, applies gofmt, and writes them to w.
+func writeGo(w io.Writer, pkg, header string, b []byte) (n int, err error) {
+	src := []byte(header)
+	src = append(src, fmt.Sprintf("package %s\n\n", pkg)...)
```

Comment:
what you can also consider is using a template file: https://pkg.go.dev/text/template
templates support loops, nested object and maybe it could make this generator simpler

sample template file is for example for a [otel readme](https://github.com/elastic/elastic-agent/blob/main/internal/pkg/otel/templates/README.md.tmpl) generated [here](https://github.com/elastic/elastic-agent/blob/883c6867e62d1b1eec3534568dc8c7c76bbb8ced/magefile.go#L3038)